### PR TITLE
Experiment with getting manually triggered workflow builds to optionally do a release

### PR DESCRIPTION
**Pull request by brad-linden**
Screenshot 2024-05-15 at 2 54 12 PM

### CODE REVIEW

**Review by nat-goodspeed — APPROVED** [review]

**Comment by nat-goodspeed** [comment]
@vir-linden , check this out!

**Comment by nat-goodspeed** [comment]
image

**Comment by brad-linden** [comment]
we may also want to add a string field or combo box or something for configuring the channel for manually triggered workflow_dispatch events

**Comment by brad-linden** [comment]
documentation for using inputs is here:
https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -22,7 +27,7 @@ jobs:
       # When you want to use a string variable as a workflow YAML boolean, it's
       # important to ensure it's the empty string when false. If you omit || '',
       # its value when false is "false", which is interpreted as true.
-      RELEASE_RUN: ${{ github.ref_type == 'tag' && startsWith(github.ref_name, 'Second_Life') && 'Y' || '' }}
+      RELEASE_RUN: ${{ (github.event.inputs.release_run || github.ref_type == 'tag' && startsWith(github.ref_name, 'Second_Life')) && 'Y' || '' }}
```
Comment (nat-goodspeed) [inline .github/workflows/build.yaml:30]: I think (but am not sure, in workflow YAML expression land) that `&&` binds more closely than `||`. That means your parens are correct as written -- but might be clearer to the casual reader if you additionally parenthesized the original `(... == 'tag' && startsWith(...))` expression.

**Review by nat-goodspeed — APPROVED** [review]
I like this change.

**Comment by vir-linden** [comment]
This is very cool, thanks!

**Comment by brad-linden** [comment]
so, unfortunately it looks like more work will be needed to make this checkbox actually functional.  currently it is failing to run the release action when a tag is not present

**Comment by nat-goodspeed** [comment]
Ah - like [this run](https://github.com/secondlife/viewer/actions/runs/9117271875/job/25069778581).

Also, I think the logic isn't quite correct yet: when I asked to run [this build](https://github.com/secondlife/viewer/actions/runs/9117271875/job/25067737711), I did _not_ check the `release_run` checkbox.

**Pull request merged**